ONSdigital/eq-author-app
# 1520 pipe unit answer types into questions

### What is the context of this PR?

here we are adding units to piping in Author and publisher V2

### How to review 

* create a new questionnaire that has a unit answer and select a unit type in the right hand panel - eg 'Metres'

- create a 2nd question after this and 'pipe the unit answer into the Question title

- make sure you have eq-survey runner spun up using the following:
**_Banch_** - 1520 pipe unit answer types into questions survey runner 
then, ensure it can be viewed in Runner by pressing the **view survey** button

**to double check the formatting of the unit piping:**

- browse to http://localhost:9000/publish/+YOUR_QUESTOINNAIRE_ID
in the resulting JSON you should see your piping question in the following layout:
{{format_unit(‘Metres’, answers[‘answerYOUR_ANSWER|_ID’])}}

- copy the resultant JSON to Postman where you will send the following request:
localhost:9000/publish/validate/+YOUR_QUESTOINNAIRE_ID

if you have no validation errors - then all is gooD!

once you are happy with this PR - please give eq-survey-runner a tick too as it's part of this PR but a seperate repo - ta

https://github.com/ONSdigital/eq-survey-runner/pull/2559

### What to do after everything is green

1. * [ ] Bring this branch up-to-date with Origin/Master
2. * [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. * [ ] Click the **Merge** button on this pull request
4. * [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. * [ ] Move the Trello card for this task into the next stage of the process
